Regex lookaround information updated

SetTrend opened the pull request:
Information in the regular expression grouping table added as discussed in https://github.com/dotnet/docs/issues/22282#issuecomment-764762593

## Summary

The original list of regular expression grouping constructs was misleading. The samples didn't convey the peculiarities of each grouping construct. Without further reference information the information in the table wasn't helpful.

In the new version the table was updated as described in issue #22282. Significant samples have been provided and brief additional information regarding _lookaround_ and _atomic_ grouping constructs has been added.

I'm not sure if I met all the coding style guides required for the docs. Please advise if something needs improvement.

Fixes #22282

dnfadmin commented:
[![CLA assistant check](https://cla.dotnetfoundation.org/pull/badge/not_signed)](https://cla.dotnetfoundation.org/dotnet/docs?pullRequest=22520) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/docs?pullRequest=22520) before we can accept your contribution.<br/><br/>:x: SetTrend [sign now](https://cla.dotnetfoundation.org/dotnet/docs?pullRequest=22520)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/docs?pullRequest=22520) it.</sub>

SetTrend commented:
@adegeo:

I have updated the regular expression grouping construct table as discussed. Please advise if things need to be amended.

adegeo commented:
Thank you @SetTrend for the effort here! The table however seems to have been changed to an HTML table which makes it harder to maintain/review. Can you swap style back?

adegeo reviewed (changes requested):
Thank you @SetTrend for the effort here! The table however seems to have been changed to an HTML table which makes it harder to maintain/review. Can you swap style back?

SetTrend commented:
Hi @adegeo:

I had to swap to HTML table because Markdown doesn't allow row spans. See my first comment in the original GitHub issue:

> In order to provide two row-spanning columns I had to use the HTML table format in favour of the Markdown format because Markdown doesn't support `rowspan`.

How do you want me to proceed?

SetTrend commented:
What I did to check for correctness was to paste the page source into a GitHub comment text editor windows (like this one). When clicking on `Preview`, you can perfectly see the result (except for API references, of course).

adegeo commented:
Ahh I see. I didn't notice it. In general we don't do row spanning just because our markdown syntax doesn't (currently) support this. We've moved on to using tables sparingly and only when we need to. For maintainability, we should keep it as markdown (As ugly as that too can be...)

There really isn't a great solution... What about having a full row and repeating the two columns?

SetTrend commented:
Sure. Let me update the table. I'll reach out here, say, tomorrow, with an update.

adegeo commented:
No worries, thank you for your contributions!!

SetTrend commented:
Hello, @adegeo,

I now updated the table so it will have the Markdown format.

I added two rows of `&mdash;` lines to separate the two similar lookaround variations:

![table](https://user-images.githubusercontent.com/9283914/106677330-df51cd00-65b8-11eb-919e-aeee752c1990.png)
<br/>

Would you feel comfortable with this version? Or what would you like me to amend?

Cheers,
Axel

SetTrend committed:
Regex lookaround information updated

Information in the regular expression grouping table added as discussed in https://github.com/dotnet/docs/issues/22282#issuecomment-764762593

SetTrend commented:
Hello @adegeo, would you mind kindly approving my pull request?

adegeo commented:
I'll get on this sorry for the delay. We've been having build issues.

adegeo commented on docs/standard/base-types/regular-expression-language-quick-reference.md:
```diff
@@ -92,11 +92,25 @@ Grouping constructs delineate subexpressions of a regular expression and typical
+|`(?<!` *subexpression* `)`|Zero-width negative lookbehind assertion.|`\b\w+\b(?<!.+and.+)`<br/><br/>&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;<br/><br/>`\b\w+\b(?<!.+and.*)`|`"cats"`, `"dogs"`, `"and"`<br/>in<br/>`"cats, dogs and some mice."`<br/>&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;<br/>`"cats"`, `"dogs"`<br/>in<b ...
+|`(?>` *subexpression* `)`|Atomic group.|`(?>a\|ab)c`|`"ac"` in`"ac"`<br/><br/>_nothing_ in`"abc"`|
+
+**Lookarounds At A Glance**
```
```suggestion
### Lookarounds at a glance
```

adegeo committed:
Update docs/standard/base-types/regular-expression-language-quick-reference.md

SetTrend commented:
Hello @adegeo, thank you for taking the time.

I'm not quite sure about the current status of this PR. You pushed and resolved your requested change yourself (excellent!). However, the PR's status still reads **"Changes requested"**. It looks like the current status refers to this comment:

/dotnet/docs/pull/22520#pullrequestreview-577820505

adegeo commented:
Still working on it 😄  The builds weren't showing up and after about an hour of waiting I had moved on to something else. I'm still here though 😄

adegeo reviewed (approved):
Looks good and gets the job done. Sorry for the long turnaround on this. We really appreciate you being available when we're not sometimes! The odd thing that happened for almost 2 weeks was the build system died and we couldn't get anything through. That came online yesterday (or the day before?) but went back offline for a bit. Seems it's stable now 🎉

SetTrend commented:
Thank you so much, @adegeo, for all the great efforts you're undertaking and particularly the efforts you've put into this PR 👍👍👍

I hope the build system will get rid of its bad cough soon 😉
